### ISSUE TITLE
[Bug]: PermissionsActivity crash

### ISSUE THREAD

**Issue by myinnos** [OneSignal/OneSignal-Android-SDK#2179]
### What happened?

The crash was reported in Crashlytics.

```
 Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
       at com.onesignal.core.activities.PermissionsActivity.onRequestPermissionsResult$lambda-0(PermissionsActivity.java:16)
       at android.os.Handler.handleCallback(Handler.java:942)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8762)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```
        

### Steps to reproduce?

```Markdown
Unfortunately, since it's a crash reported in Crashlytics, I haven't been able to reproduce.
```

### What did you expect to happen?

No crash

### OneSignal Android SDK version

5.1.20

### Android version

14, 13

### Specific Android models

```Markdown
Samsung, Xiaomi
```

### Relevant log output

```Shell
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
       at com.onesignal.core.activities.PermissionsActivity.onRequestPermissionsResult$lambda-0(PermissionsActivity.java:16)
       at android.os.Handler.handleCallback(Handler.java:942)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8762)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jinliu9508** [comment]
@myinnos Thank you for reporting this issue. We have noticed this crash occurring in versions 5.1.18 - 5.1.20 and are currently working on a fix. We will notify you once the fix is available. In the meantime, if you have encountered this crash during your development, please share your steps to reproduce it if that's possible.

**Comment by programmerpinggiran** [comment]
i also got this error.

**Comment by UgurGuner** [comment]
Several users also experienced the same crash in my app. Here is the one example stack trace:

ArrayIndexOutOfBoundsException
length=0; index=0
mechanism
UncaughtExceptionHandler
handled
false

com.onesignal.core.activities.PermissionsActivity in onRequestPermissionsResult$lambda-0 at line 101
com.onesignal.core.activities.PermissionsActivity in $r8$lambda$ECZClNxyXcb1TMRbSJOUwr-2fOM
com.onesignal.core.activities.PermissionsActivity$$ExternalSyntheticLambda0 in run
android.os.Handler in handleCallback at line 958
android.os.Handler in dispatchMessage at line 99
android.os.Looper in loopOnce at line 230
android.os.Looper in loop at line 319
android.app.ActivityThread in main at line 8919
java.lang.reflect.Method in invoke
com.android.internal.os.RuntimeInit$MethodAndArgsCaller in run at line 578
com.android.internal.os.ZygoteInit in main at line 1103

Device: Family
SM-S911B

**Comment by amrutkarsuraj** [comment]
In Firebase cryashlyatics show this error-

Fatal Exception: java.lang.ArrayIndexOutOfBoundsException
length=0; index=0
com.onesignal.core.activities.PermissionsActivity.onRequestPermissionsResult$lambda-0 (PermissionsActivity.kt:101)

device info 
Poco C61
Version:Android 14
![image](https://github.com/user-attachments/assets/e1bc10d3-c088-4847-8afd-b73dd5362d09)

**Comment by michael-winkler** [comment]
@jinliu9508 After two weeks still no fix 🙄

**Comment by fernandodrummond** [comment]
Facing this issue too

**Comment by jinliu9508** [comment]
This issue has been fixed in v5.1.21. Please upgrade to our latest SDK version and let us know if you encounter any issues. Thank you for your patience and support!
